diff --git a/src/Perception/src/ArucoDetector.cpp b/src/Perception/src/ArucoDetector.cpp
index 3b05fdcf96..b07d987279 100644
--- a/src/Perception/src/ArucoDetector.cpp
+++ b/src/Perception/src/ArucoDetector.cpp
@@ -55,6 +55,11 @@ class ArucoDetector::Impl
     /**
      * Utility map for choosing Aruco marker dictionary depending on user parameter
      */
+    // TODO(traversaro): when we drop support for OpenCV < 4.7.0, we can cleanup this part
+    // we can also check if there are other dictionary that should be added here
+#if (CV_VERSION_MAJOR >= 5) || (CV_VERSION_MAJOR == 4 && CV_VERSION_MINOR >= 7)
+#define PREDEFINED_DICTIONARY_NAME PredefinedDictionaryType
+#endif
     std::unordered_map<std::string, cv::aruco::PREDEFINED_DICTIONARY_NAME>
         availableDict{{"4X4_50", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_4X4_50},
                       {"4X4_100", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_4X4_100},
@@ -115,7 +120,14 @@ bool ArucoDetector::initialize(std::weak_ptr<const IParametersHandler> handler)
         return false;
     }
 
+// In OpenCV 4.7.0 getPredefinedDictionary started returning a cv::aruco::Dictionary
+// instead of a cv::Ptr<cv::aruco::Dictionary>
+#if (CV_VERSION_MAJOR >= 5) || (CV_VERSION_MAJOR == 4 && CV_VERSION_MINOR >= 7)
+    m_pimpl->dictionary = cv::makePtr<cv::aruco::Dictionary>();
+    *(m_pimpl->dictionary) = cv::aruco::getPredefinedDictionary(m_pimpl->availableDict.at(dictName));
+#else
     m_pimpl->dictionary = cv::aruco::getPredefinedDictionary(m_pimpl->availableDict.at(dictName));
+#endif
 
     if (!handle->getParameter("marker_length", m_pimpl->markerLength))
     {
