diff --git a/src/ReducedModelControllers/src/CentroidalMPC.cpp b/src/ReducedModelControllers/src/CentroidalMPC.cpp
index ef265ed464..81d05145ba 100644
--- a/src/ReducedModelControllers/src/CentroidalMPC.cpp
+++ b/src/ReducedModelControllers/src/CentroidalMPC.cpp
@@ -940,7 +940,7 @@ struct CentroidalMPC::Impl
                         0 <= casadi::MX::mtimes(casadi::MX::reshape(contact.orientation(Sl(), i),
                                                                     3,
                                                                     3),
-                                                corner.force(Sl(), i)(2)));
+                                                corner.force(Sl(), i))(2));
                 }
             }
         }
